File: mobile-swift/DishMatch/Models/Restaurant.swift
import Foundation

struct Restaurant: Codable, Identifiable {
    let id: UUID
    let googlePlaceId: String
    let name: String
    let cuisineTags: [String]
    let priceTier: String?
    let rating: Double?
    let photoUrl: String?
    let address: String?
    let lat: Double
    let lng: Double
    let description: String?
    let reviews: [String]?
    let vibeBlurb: String?
    let overallVibeQuotes: [String]?
}

struct VibePick: Codable {
    let restaurant: Restaurant
    let narrative: String
}

struct PersonalizedFit: Codable {
    let dietaryMatch: Bool
    let budgetMatch: Bool
    let cuisineOverlap: [String]
    let narrative: String
}

struct PersonalizedFitContext: Identifiable {
    let id: UUID
    let restaurantName: String
    let fit: PersonalizedFit
}

struct SessionResult: Codable, Identifiable {
    let restaurant: Restaurant
    let scorePct: Double
    let yesCount: Int
    let total: Int

    var id: UUID { restaurant.id }
}

File: backend/models/restaurant.py
from uuid import UUID, uuid4

from beanie import Document
from pydantic import Field
from pymongo import IndexModel


class Restaurant(Document):
    id: UUID = Field(default_factory=uuid4)
    google_place_id: str
    name: str
    cuisine_tags: list[str] = Field(default_factory=list)
    price_tier: str | None = None  # "$" | "$$" | "$$$" | "$$$$"
    rating: float | None = None
    photo_url: str | None = None
    address: str | None = None
    lat: float
    lng: float
    description: str | None = None  # editorial_summary from Places Details (cached, single API call)
    reviews: list[str] | None = None          # top 3 review text snippets from Places Details
    vibe_blurb: str | None = None             # Gemini-generated atmosphere summary
    overall_vibe_quotes: list[str] | None = None  # 2-3 short quotes Gemini picks from reviews

    class Settings:
        name = "restaurants"
        indexes = [IndexModel("google_place_id", unique=True)]
